src/common/snippets/src/op/subgraph.cpp

```diff
@@ -388,27 +412,30 @@ ov::PartialShape snippets::op::Subgraph::canonicalize(const BlockedShapeVector&
     } else {
         for (size_t i = 0; i < body_results.size(); i++) {
             auto shape_i = body_results[i]->get_input_partial_shape(0);
-            auto outputShape_i = std::get<0>(outputShapes[i]);
-            // Check that the produced output shape corresponds to the passed shape
-            // Some produced shapes may have been changed to be broadcastable (e.g. blocked + planar outputs),
-            // so we need to remove leading and trailing "1" before the comparison
-            PartialShape pShape_i(skipStartEndOnes(shape_i));
-            bool compatibleWithPassedShape = PartialShape::broadcast_merge_into(pShape_i,
-                                                                                skipStartEndOnes(outputShape_i),
-                                                                                ::ov::op::AutoBroadcastType::NUMPY);
-            NODE_VALIDATION_CHECK(this, compatibleWithPassedShape,
-                                  "Inferred and passed results shapes are incompatible for snippet ");
+            if (!reshape) {
+                auto outputShape_i = std::get<0>(outputShapes[i]);
+                // Check that the produced output shape corresponds to the passed shape
+                // Some produced shapes may have been changed to be broadcastable (e.g. blocked + planar outputs),
+                // so we need to remove leading and trailing "1" before the comparison
+                PartialShape pShape_i(skipStartEndOnes(shape_i));
+                bool compatibleWithPassedShape = PartialShape::broadcast_merge_into(pShape_i,
+                                                                                    skipStartEndOnes(outputShape_i),
+                                                                                    ::ov::op::AutoBroadcastType::NUMPY);
+                NODE_VALIDATION_CHECK(this, compatibleWithPassedShape,
+                                      "Inferred and passed results shapes are incompatible for snippet ");
+            }
             // Check that output shapes are broadcastable to each other => can be scheduled
             bool compatibleWithOtherOutputs = PartialShape::broadcast_merge_into(outPShape, shape_i,
                                                                                  ::ov::op::AutoBroadcastType::NUMPY);
             NODE_VALIDATION_CHECK(this, compatibleWithOtherOutputs,
                                   "Snippets output shapes must be numpy broadcastable");
         }
     }
-
-    // We should insert Converts after Parameters and Constant and before Results
-    // to align precision inside Subgraph body that is supported by Plugin
-    align_element_types(outputShapes, inputShapes);
+    if (!reshape) {
```

Comment:
Honestly, it's not clear why if not `reshape`, we shouldn't insert Converts (I understand but another developer without background will be confused).
On the [previous](https://github.com/openvinotoolkit/openvino/pull/15430#discussion_r1230494695) comment I meant to unite only plugin methods but to leave the Snippets so because now the method `snippets::op::Subgraph::canonicalize` looks complicated now. What do you think?

Reply:
I agree, created separate snippets function canonicalized_body_shape_infer() for reshape case. thanks Alexandra!